pkg/gui/controllers/helpers/window_arrangement_helper.go

```diff
@@ -203,31 +213,62 @@ func (self *WindowArrangementHelper) infoSectionChildren(informationStr string,
+	// between at the end
+	var result []*boxlayout.Box
 
-	if !self.c.UserConfig.Gui.ShowBottomLine {
-		optionsBox.Weight = 0
-		appStatusBox.Weight = 1
-	} else {
-		optionsBox.Weight = 1
-		if self.c.InDemo() {
-			// app status appears very briefly in demos and dislodges the caption,
-			// so better not to show it at all
-			appStatusBox.Size = 0
-		} else {
-			appStatusBox.Size = runewidth.StringWidth(INFO_SECTION_PADDING) + runewidth.StringWidth(appStatus)
+	if !self.c.InDemo() {
+		// app status appears very briefly in demos and dislodges the caption,
+		// so better not to show it at all
+		if appStatus != "" {
+			result = append(result, &boxlayout.Box{Window: "appStatus", Size: runewidth.StringWidth(appStatus)})
 		}
 	}
 
-	result := []*boxlayout.Box{appStatusBox, optionsBox}
+	if self.c.UserConfig.Gui.ShowBottomLine {
+		result = append(result, &boxlayout.Box{Window: "options", Weight: 1})
+	}
 
 	if (!self.c.InDemo() && self.c.UserConfig.Gui.ShowBottomLine) || self.modeHelper.IsAnyModeActive() {
-		result = append(result, &boxlayout.Box{
-			Window: "information",
-			// unlike appStatus, informationStr has various colors so we need to decolorise before taking the length
-			Size: runewidth.StringWidth(INFO_SECTION_PADDING) + runewidth.StringWidth(utils.Decolorise(informationStr)),
-		})
+		result = append(result,
+			&boxlayout.Box{
+				Window: "information",
+				// unlike appStatus, informationStr has various colors so we need to decolorise before taking the length
+				Size: runewidth.StringWidth(utils.Decolorise(informationStr)),
+			})
+	}
+
+	if len(result) == 3 {
+		// status - options - information
+		result = slices.Insert(result, 0, spacerBox("statusSpacer1"))
+		result = slices.Insert(result, 2, spacerBox("statusSpacer2"))
+		result = slices.Insert(result, 4, spacerBox("statusSpacer3"))
+		result = slices.Insert(result, 6, spacerBox("statusSpacer4"))
+	} else if len(result) == 2 {
+		if result[0].Window == "appStatus" {
+			// status - information
+			result = slices.Insert(result, 0, spacerBox("statusSpacer1"))
+			result = slices.Insert(result, 2, spacerBox("statusSpacer2"))
+			result = slices.Insert(result, 3, flexibleSpacerBox("statusSpacer3"))
+			result = slices.Insert(result, 5, spacerBox("statusSpacer4"))
+		} else {
+			// options - information
+			result = slices.Insert(result, 0, spacerBox("statusSpacer1"))
+			result = slices.Insert(result, 2, spacerBox("statusSpacer2"))
+			result = slices.Insert(result, 4, spacerBox("statusSpacer3"))
+		}
+	} else if len(result) == 1 {
+		if result[0].Window == "information" {
+			// information
+			result = slices.Insert(result, 0, flexibleSpacerBox("statusSpacer1"))
+			result = slices.Insert(result, 2, spacerBox("statusSpacer2"))
+		} else {
+			// status
+			result[0].Size = 0
+			result[0].Weight = 1
+			result = slices.Insert(result, 0, spacerBox("statusSpacer1"))
+			result = slices.Insert(result, 2, spacerBox("statusSpacer2"))
+		}
```

Comment:
There's probably a more elegant way to write this code...

Reply:
I agree: we could have a function which takes a result and inserts spacer boxes between each element. Then that just leaves the flexible spacer boxes, which I'm happy to be inserted directly. It would be good to add comments explaining why the flexible spacer boxes are inserted when they are

Reply:
Here's one way of doing that: bb54d7c273ed. It's a little bit less code, but I'm undecided whether it makes it easier to understand. You decide. 😄  (Also feel free to make or suggest further improvements.)

Reply:
I've got some tweaks that I'm happy to push up but I want to ask: do we really want this line to have one character of padding on the left/right side? I actually prefer it just taking up all available horizontal space (and tig/gitui do the same)

Reply:
Right now (on master) we do have a space to the left of the options view and to the right of the information view. I think it looks good this way. The fact that the app status and the search/filter view don't use the same alignment always felt like a bug to me.

Reply:
As discussed in town hall let's just not have any outer padding